browser/js/modalWindow/modal.controller.js

```diff
@@ -0,0 +1,28 @@
+app.controller('ModalInstanceCtrl', function ($scope, $uibModalInstance, ModalFactory) {
+
+  // $scope.items = items;
+  // $scope.selected = {
+  //   item: $scope.items[0]
+  // };
+
+  $scope.ok = function () {
+    $uibModalInstance.close();
+  };
+
+  $scope.cancel = function () {
+    $uibModalInstance.dismiss('cancel');
+  };
+
+  $scope.post= function(){
+    $scope.kit.categories= $scope.kit.categories.split(",");
+    ModalFactory.postKit($scope.kit)
+    .then(()=> {
+     $scope.ok();
+     // $scope.$digest();  how do i make the digest work upon adding a new page?
```

Comment:
My cart was having a similar problem and, (with @jdhang's suggestion) I ended up using $state.go('cart') - which did the digest for me.  Could that work here?  Like you can redirect to the same "kit" state maybe, but this time it will be updated?